.github/workflows/pr-build.yml

```
@@ -0,0 +1,43 @@
+name: PR build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [ main ]
+  merge_group:
+
+    
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+  
+  
+jobs:
+  checks:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: jverify-collection
+          submodules: recursive
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: '21'
+          cache: gradle
+      - name: Build Dafny
+        working-directory: ./jverify-collection/dafny
+        run: |
+          dotnet build ./Source/Dafny.sln
+          make z3-ubuntu
+      - name: Code generation has no diff
+        working-directory: ./jverify-collection
+        run: |
+          make test-generation-nodiff
+      - name: Install Z3
+        run: |
```

I think you meant to write this?

```suggestion
      - name: Build Dafny
        working-directory: ./jverify-collection/dafny
        run: |
          dotnet build ./Source/Dafny.sln
      - name: Install Z3
        working-directory: ./jverify-collection/dafny
        run: |
          make z3-ubuntu
      - name: Code generation has no diff
        working-directory: ./jverify-collection
        run: |
          make test-generation-nodiff
```